bundles/AdminBundle/Resources/public/js/pimcore/document/document.js

```diff
@@ -103,7 +103,21 @@ pimcore.document.document = Class.create(pimcore.element.abstract, {
 
             }
 
-            pimcore.plugin.broker.fireEvent("preSaveDocument", this, this.getType(), task, only);
+            try {
+                pimcore.plugin.broker.fireEvent("preSaveDocument", this, this.getType(), task, only);
+            } catch (e) {
+                if (e instanceof pimcore.error.ValidationException) {
+                        this.tab.unmask();
+                        pimcore.helpers.showPrettyError('document', t("error"), e.message);
```

Comment:
just like objects...please use t("saving_failed") for text(3rd param) and e.message for errorText(4th param)

Reply:
fixed